### PR TITLE
remove MostRequestedPriority from doc since already deprecated

### LINKED ISSUE #668: MostRequestedPriority is still mentioned in documentation, but seems to be gone many releases ago
The documentation of strategy HighNodeUtilization states that it must be used in conjunction with MostRequestedPriority.

But MostRequestedPriority is gone and nowhere documented. It would suggest to remove it from the docs

### PR DESCRIPTION
**Pull request by JaneLiuL**
fix #668

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -274,8 +274,7 @@ under utilized frequently or for a short period of time. By default, `numberOfNo
 This strategy finds nodes that are under utilized and evicts pods from the nodes in the hope that these pods will be 
 scheduled compactly into fewer nodes.  Used in conjunction with node auto-scaling, this strategy is intended to help 
 trigger down scaling of under utilized nodes.
-This strategy **must** be used with the scheduler strategy `MostRequestedPriority`. The parameters of this strategy are 
-configured under `nodeResourceUtilizationThresholds`.
+The parameters of this strategy are configured under `nodeResourceUtilizationThresholds`.
```
Comment (damemi) [inline README.md:277]: @JaneLiuL it's still important to mention that this has to be used with the right scheduler config, like in https://github.com/kubernetes-sigs/descheduler/issues/668#issuecomment-988167924 and https://github.com/kubernetes-sigs/descheduler/issues/668#issuecomment-989887089

File: docs/proposals.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Proposals
```
Comment (damemi) [inline docs/proposals.md:1]: is this related to removing MostRequestedPriority?

File: docs/user-guide.md [hunk 3]
```diff
@@ -118,7 +118,7 @@ strategies:
 
 #### Balance low utilization nodes
 Using `HighNodeUtilization`, descheduler will rebalance the cluster based on memory by evicting pods
-from nodes with memory utilization lower than 20%. This should be used along with scheduler strategy `MostRequestedPriority`.
+from nodes with memory utilization lower than 20%.
```
Comment (damemi) [inline docs/user-guide.md:121]: same as above, we don't just want to remove MostRequestedPriority; need to update with the new change

**Review by damemi — COMMENTED** [review]
/ok-to-test

File: docs/proposals.md (hunk 2 above)
Comment (JaneLiuL) [reply docs/proposals.md:1]: no, it is related to https://github.com/kubernetes-sigs/descheduler/issues/646, i just thinking that for use one pr for fix two document related issues. is it change the pr title will be better?

Comment (damemi) [reply docs/proposals.md:1]: technically if it's unrelated it should probably be a different PR, to organize them better. Hope you don't mind

Comment (JaneLiuL) [reply docs/proposals.md:1]: get it, i don't mind :) i will put this on a different pr :)

**Comment by damemi** [comment]
/approve
/lgtm
thanks @JaneLiuL !
